test/RibbonV2SharePrice.test.ts

```diff
@@ -0,0 +1,200 @@
+// --- Imports ---
+import { artifacts, ethers, network, waffle } from 'hardhat';
+import { expect } from 'chai';
+import type { BigNumberish } from '@ethersproject/bignumber';
+import { keccak256 } from '@ethersproject/keccak256';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address';
+import { MockCozyToken, IRibbonVaultV2, RibbonV2SharePrice, IERC20 } from '../typechain';
+
+// --- Constants and extracted methods ---
+const { deployContract, loadFixture } = waffle;
+const { MaxUint256: MAX_UINT } = ethers.constants;
+const { defaultAbiCoder, hexZeroPad, hexStripZeros } = ethers.utils;
+const BN = (x: BigNumberish) => ethers.BigNumber.from(x);
+const to32ByteHex = (x: BigNumberish) => hexZeroPad(BN(x).toHexString(), 32);
+
+interface VaultInfo {
+  name: string;
+  ribbonVault: string;
+}
+
+// Ribbon vault pairings to test
+const vaults: VaultInfo[] = [
+  {
+    name: 'T-ETH-C',
+    ribbonVault: '0x25751853Eab4D0eB3652B5eB6ecB102A2789644B',
+  },
+];
+
+// --- Storage slot helper methods ---
+// `defaultAbiCoder.encode` is equivalent to Solidity's `abi.encode()`, and we strip leading zeros from the hashed
+// value to conform to the JSON-RPC spec: https://ethereum.org/en/developers/docs/apis/json-rpc/#hex-value-encoding
+
+// Returns the storage slot for a Solidity mapping from an `address` to a value, given the slot of the mapping itself,
+//  `mappingSlot`. Read more at https://docs.soliditylang.org/en/latest/internals/layout_in_storage.html#mappings-and-dynamic-arrays
+const getSolidityStorageSlot = (mappingSlot: string, address: string) => {
```

Comment:
We probably should pull these into a `utils.ts` file since we've copy/pasted them a lot, but no need to do that in this PR